src/lib/helpers.ts

```diff
@@ -18,24 +21,79 @@ export function makeObject(array: any[], arrayKey: string): any {
 export function makeArray(object: any, objectKey: string): any[] {
   return Object.keys(object).reduce((acc: any[], key: string) => {
     if (objectKey in object[key]) {
-      acc.push({ ...object[key], [objectKey]: makeArray(object[key][objectKey], objectKey) })
+      acc.push({
+        ...object[key],
+        [objectKey]: makeArray(object[key][objectKey], objectKey),
+      })
       return acc
     }
     acc.push(object[key])
     return acc
   }, [])
 }
 
-export function paths(object: any, prev: string = ''): Path[] {
+export function paths(
+  object: any,
+  prev = '',
+  type: Path['type'] = 'text'
+): Path[] {
   if (object) {
     return Object.keys(object).reduce((acc: any[], key: string) => {
       const path = `${prev}${prev ? `.${key}` : key}`
       const value = object[key]
+      let valueType = type
+
+      if (!isNaN(Date.parse(value))) {
+        valueType = 'date'
+      }
+
+      if (typeof value === 'boolean') {
+        valueType = 'boolean'
+      }
+
+      if (
+        valueType === 'text' &&
+        Array.isArray(value) &&
+        typeof value[0] === 'object' &&
+        isStructuredTextSlate(value[0])
+      ) {
+        valueType = 'structured_text'
+      }
+
+      if (valueType === 'text' && typeof value === 'object' && isColor(value)) {
+        valueType = 'color'
+      }
+
+      if (valueType === 'text' && typeof value === 'object' && isImage(value)) {
+        valueType = 'media'
+      }
+
+      if (
+        key === 'itemTypeId' ||
+        key === 'itemId' ||
+        key === 'upload_id' ||
+        key === 'id'
+      ) {
+        valueType = 'id'
+      }
+
+      if (Number(value)) {
+        valueType = 'number'
+      }
+
+      if (htmlRegex.test(value)) {
```

Comment:
Guess this check is last since otherwise having 'html-like' strings in the value would cause a false positive?